File: src/main/org/nlogo/window/PeriodicUpdater.java
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.window;

import org.nlogo.api.PeriodicUpdateDelay;
import org.nlogo.nvm.JobManagerInterface;

public strictfp class PeriodicUpdater
    extends javax.swing.Timer
    implements java.awt.event.ActionListener {

  private final JobManagerInterface jobManager;

  public PeriodicUpdater(JobManagerInterface jobManager) {
    super(PeriodicUpdateDelay.PERIODIC_UPDATE_DELAY(), null);
    this.jobManager = jobManager;
    addActionListener(this);
  }

  public void actionPerformed(java.awt.event.ActionEvent e) {
    jobManager.timeToRunSecondaryJobs();
  }

}

File: src/main/org/nlogo/window/PeriodicUpdater.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.window

import java.awt.event.{ ActionEvent, ActionListener }
import javax.swing.Timer
import org.nlogo.api.PeriodicUpdateDelay
import org.nlogo.nvm.JobManagerInterface

class PeriodicUpdater(jobManager: JobManagerInterface)
    extends Timer(PeriodicUpdateDelay.PERIODIC_UPDATE_DELAY, null) with ActionListener {
  addActionListener(this)
  def actionPerformed(e: ActionEvent): Unit = jobManager.timeToRunSecondaryJobs()
}
